### PR TITLE
cloud ide instructions

### PR DESCRIPTION
**Pull request by tteltrab**
Initial first pass at cloud IDE instructions.

Decided to go with Codenvy instructions for now (can add Cloud9 later, if necessary). It's the best balance of ease-of-use + free (no credit cards required, like with Cloud9).

### CODE REVIEW

**Comment by tteltrab** [comment]
Note: prior to merge, we'll want to change:

```sh
curl https://raw.githubusercontent.com/university-bootcamp/coding-environment/cloud-ide-instructions/github_auth.rb > ~/.bootcamp-github.rb && ruby ~/.bootcamp-github.rb
```

to

```sh
curl https://raw.githubusercontent.com/university-bootcamp/coding-environment/master/github_auth.rb > ~/.bootcamp-github.rb && ruby ~/.bootcamp-github.rb
```

File: cloud-ide.md [hunk 1]
```diff
@@ -1 +1,267 @@
...
+ 
+### Step 2: Setting up the Database (Postgres)
+
+Next, we need to set up our application to use postgres:
+
+1. Ensure postgres is running:
+    ```sh
+    service postgresql start
+    ```
+2. Open up the `config/database.yml` file in your editor and change it so that it looks exactly like the below:
```
Comment (kenmazaika) [inline cloud-ide.md:197]: maybe we can make quick note that these files are senstitive to whitespace?  I common problem when editting the database.yml is to change whitespace.

File: cloud-ide.md [hunk 2]
```diff
@@ -1 +1,267 @@
...
+We can use `rails new` to create the new app, though we'll need to make sure we create the app in the right folder.
+
+1. For us to be able to see the code in the editor, we'll need to be in projects folder any time we create a project. In the terminal, type the following: 
+    ```sh
+    cd projects
+    ```
+
+2. We can create a new rails app, install dependencies (done as part of `rails new`), and move into our new project folder using the following commands:
+   ```sh
+   rails new test2
```
Comment (kenmazaika) [inline cloud-ide.md]: you should use the command:

```
rails new rails-test-app -d postgresql
```

To create a project called `rails-test-app` and do the database configuration in a single step.  We use this in the other guides.

File: cloud-ide.md [hunk 3]
```diff
@@ -1 +1,267 @@
...
+
+6. Click the link to see the resulting application!
+
+   <img width="700" alt="codenvy resulting app" src="images/codenvy/project-setup/5-result.png" />
+
+Congratulations, your application is now up and running!
+
+If you want to stop your running application, simply close the rails process in the terminal/process pane.
+
+> Note: Remember <a href="#workspace-note">this note regarding how your workspace will reset after it has been stopped</a>. Also remember to re-navigate into your project using `cd projects/splurty`.
```
Comment (kenmazaika) [inline cloud-ide.md]: This is the instruction to how to use the `splurty`, but I think maybe it should be generic?

File: mac.md [hunk 4]
```diff
@@ -93,7 +93,11 @@ The following information will be important for you to know about the installati
 
 **If you ever want to preview the application that is running within your coding environment** — you will need to...
 
-[STEPS HERE]
+  - Use the `Run` button provided in the browser environment. The IDE will provide you with a URL you can use to view your application:
+
+    <img width="600" alt="codenvy run button" src="images/codenvy/run.png" />
```
Comment (kenmazaika) [inline mac.md]: Since we don't have other screenshots in these high level OS guides, I think to make it consistent, it might be best to remove this (as opposed to adding screenshots to other environments).  Just my 2-cents - open to thoughts here.

File: windows.md [hunk 5]
```diff
@@ -93,7 +93,11 @@ The following information will be important for you to know about the installati
 
 **If you ever want to preview the application that is running within your coding environment** — you will need to...
 
-[STEPS HERE]
+  - Use the `Run` button provided in the browser environment. The IDE will provide you with a URL you can use to view your application:
+
+    <img width="600" alt="codenvy run button" src="images/codenvy/run.png" />
```
Comment (kenmazaika) [inline windows.md]: Same as above.

File: cloud-ide.md [hunk 6]
```diff
@@ -1 +1,267 @@
...
+
+<span id="stacks">**Stacks**</span> define what programming languages, programs, and commands should be available on the "machine" your workspace is running on. Setting up a stack is the equivalent of making a list of software you need to install on your laptop.
+
+Follow these steps to set up the stack we'll be using:
+
+1. Click **Stacks** in the left navigation, and then click **Build Stack From Recipe**.
+
+   <img width="350" alt="Create Stack Step 1" src="images/codenvy/account-setup/6.png" />
+
+2. Paste `FROM tteltrab/rr-pg-for-codenvy` into the box, and click **OK**.
```
Comment (kenmazaika) [inline cloud-ide.md]: I think we might want to make this a little more explicit what the steps are.

2. Find the the textbox that is presented for the x field and enter the following code in there:

```
FROM tteltrab/rr-pg-for-codenvy
```

Then click the `OK` button.

Reason being, when I first read this I wasn't sure if there was a separate thing that I should copy and paste from (as opposed to this specific value).

File: cloud-ide.md [hunk 7]
```diff
@@ -1 +1,267 @@
...
+
+   <img width="350" alt="Create Stack Step 2" src="images/codenvy/account-setup/7.png" />
+
+3. Name your stack `bootcamp`, give it a description, and click **Save**
```
Comment (kenmazaika) [inline cloud-ide.md]: We talked about this a little on slack.  I think we should make a note about making it unique, since it will have to be to pass validations.

**Review by kenmazaika — COMMENTED** [review]
I just made a couple small nitpicks.  Wanted to give my feedback, happy to hit the merge button, though if we trust the QA process for this.  I know from standup today you mentioned that there was QA being done for this, but not sure if this is finalized yet.

Looks great overall!

File: mac.md (hunk 4 above)
Comment (tteltrab) [reply mac.md]: That's fair - I think it makes it easier, but linking to the guide is probably sufficient.

File: windows.md (hunk 5 above)
Comment (tteltrab) [reply windows.md]: https://github.com/university-bootcamp/coding-environment/pull/4#discussion_r232346194

**Comment by kenmazaika** [comment]
@tteltrab I think accounting for `Name your stack` needing a unique name is the only open change, are there any other changes you want to make?

I'm happy to hit the merge button now if you think we should keep things like this.  Just wanting to flag it since it looks like it's not changed unless I'm missing something and get the green light from you before hitting the merge button.

**Comment by tteltrab** [comment]
@kenmazaika - I added a note here in the last commit: https://github.com/university-bootcamp/coding-environment/pull/4/files#diff-60751ed994160533e43b3d3e25e27640R83

I was thinking that + the UI telling you it's a problem would be sufficient, were you thinking something else?

**Comment by kenmazaika** [comment]
I'm open to anything, but my thought was this is a little misleading, since this won't actually work:

> Name your stack `bootcamp`, give it a description, and click **Save**

I can see the error coming up and flagging it for TA help before someone skips ahead to read the note.

Maybe something more like:

> Enter a name for your stack, but keep in mind that it needs to be unique.  Our suggestion is to use the format of `bootcamp-YOUR-NAME` and replace this with your name, but you can give it any name so long as you reference this name you've choosen in the next step.  Then enter a description, and click **Save*

This might be too wordy though.  If you think this isn't a good idea, happy to merge it like it is.

**Comment by tteltrab** [comment]
Sounds good. I think I misunderstood - my impression was that stacks had to have a unique name as related to your account, but I guess they have to be unique globally? I'm a bit surprised at that. 

Either way, updated.

**Comment by kenmazaika** [comment]
:+1:
